BitList.cs

```diff
@@ -0,0 +1,1553 @@
+    /// </summary>
+    public bool Remove(bool bit)
+    {
+        try
+        {
+            RemoveAt(IndexOf(bit));
+            return true;
+        }
+        catch
+        {
+            return false;
+        }
+    }
+
+    /// <summary>
+    /// Remove the specified list of bit from this instance.
+    /// </summary>
+    /// <exception cref="EmptyInstanceException" />
+    public bool Remove(BitList bits)
+    {
+        if (IsEmpty)
+            throw new EmptyInstanceException();
+        else
+        {
+            var thisBitList = ToText();
+            var thoseBitList = bits.ToText();
+            if (!thisBitList.Contains(thoseBitList))
+                return false;
+            var newBitList = thisBitList.Replace(thoseBitList, string.Empty);
+            this.bits = Parse(newBitList).bits;
+            return true;
+        }
+    }
+
+    /// <summary>
+    /// Insert the specified list of bit in the specified index.
+    /// </summary>
+    /// <exception cref="IndexOutOfRangeException" />
+    public void Insert(int index, BitList bits)
+    {
+        if (IsEmpty)
+            this.bits = new BitArray(bits.bits);
+        else if (index < bits.Count && index >= 0)
+        {
+            var bools = ToBooleanArray().ToList();
+            bools.InsertRange(index, bits.ToBooleanArray());
+            this.bits = new BitArray(bools.ToArray());
+        }
+        else
+            throw new IndexOutOfRangeException();
+    }
+
+    /// <summary>
+    /// Insert the specified bit in the specified index.
+    /// </summary>
+    /// <exception cref="IndexOutOfRangeException" />
+    public void Insert(int index, bool bit)
+    {
+        if (IsEmpty)
+            bits = new BitArray(new[] { bit });
+        else if (index < bits.Length && index >= 0)
+        {
+            var bools = ToBooleanArray().ToList();
+            bools.Insert(index, bit);
+            bits = new BitArray(bools.ToArray());
+        }
+        else
+            throw new IndexOutOfRangeException();
+    }
+
+    /// <summary>
+    /// Replace a list of bit of the current instance with another list of bit.
+    /// </summary>
+    /// <exception cref="EmptyInstanceException" />
+    public void Replace(BitList oldBitList, BitList newBitList)
+    {
+        if (IsEmpty)
+            throw new EmptyInstanceException();
+        else
+        {
+            var thisBitListText = ToText();
+            var oldBitListText = oldBitList.ToText();
+            var newBitListText = newBitList.ToText();
+            var bitsText = thisBitListText.Replace(oldBitListText, newBitListText);
+            bits = Parse(bitsText).bits;
+        }
+    }
+
+    /// <summary>
+    /// Set the value in the specified index with a new value.
+    /// </summary>
+    /// <exception cref="ArgumentOutOfRangeException" />
+    public void Set(int index, bool value)
+    {
+        if (!IsEmpty && (index < bits.Length | index >= 0))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Correct this '|' to '||'.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5545753)